tee-worker/ts-tests/integration-tests/common/di-utils.ts

```diff
@@ -257,34 +256,50 @@ export async function createSignedTrustedCallActivateIdentity(
         signer,
         mrenclave,
         nonce,
-        [subject.toHuman(), subject.toHuman(), identity, aesKey, hash]
+        [primeIdentity.toHuman(), primeIdentity.toHuman(), identity, aesKey, hash]
     );
 }
 
 export async function createSignedTrustedGetterIdGraph(
     parachainApi: ApiPromise,
     signer: Signer,
-    subject: LitentryPrimitivesIdentity
+    primeIdentity: LitentryPrimitivesIdentity
 ): Promise<Getter> {
     const getterSigned = await createSignedTrustedGetter(
         parachainApi,
         ['id_graph', '(LitentryIdentity)'],
         signer,
-        subject.toHuman()
+        primeIdentity.toHuman()
     );
     return parachainApi.createType('Getter', { trusted: getterSigned }) as unknown as Getter; // @fixme 1878;
 }
 
 export const getSidechainNonce = async (
     context: IntegrationTestContext,
     teeShieldingKey: KeyObject,
-    subject: LitentryPrimitivesIdentity
+    primeIdentity: LitentryPrimitivesIdentity
 ): Promise<Index> => {
-    const getterPublic = createPublicGetter(context.api, ['nonce', '(LitentryIdentity)'], subject.toHuman());
+    const getterPublic = createPublicGetter(context.api, ['nonce', '(LitentryIdentity)'], primeIdentity.toHuman());
     const getter = context.api.createType('Getter', { public: getterPublic }) as unknown as Getter; // @fixme 1878
-    const nonce = await sendRequestFromGetter(context, teeShieldingKey, getter);
-    const nonceValue = decodeNonce(nonce.value.toHex());
-    return context.api.createType('Index', nonceValue) as Index;
+    const res = await sendRequestFromGetter(context, teeShieldingKey, getter);
+    const nonce = context.api.createType('Option<Bytes>', hexToU8a(res.value.toHex())).unwrap();
+    return context.api.createType('Index', nonce);
+};
+
+export const getIdGraphHash = async (
+    context: IntegrationTestContext,
+    teeShieldingKey: KeyObject,
+    primeIdentity: LitentryPrimitivesIdentity
+): Promise<H256> => {
+    const getterPublic = createPublicGetter(
+        context.api,
+        ['id_graph_hash', '(LitentryIdentity)'],
+        primeIdentity.toHuman()
+    );
+    const getter = context.api.createType('Getter', { public: getterPublic }) as unknown as Getter; // @fixme 1878
+    const res = await sendRequestFromGetter(context, teeShieldingKey, getter);
+    const hash = context.api.createType('Option<Bytes>', hexToU8a(res.value.toHex())).unwrap();
```

Comment:
nitpicking: it should work without hex -> u8a 
```suggestion
    const hash = context.api.createType('Option<Bytes>', res.value.toHex()).unwrap();
```

Reply:
No it doesn't, I believe I had similar questions.

without `hexToU8a` I got:
```
Error: createType(LinkIdentityResult):: decodeU8aStruct: failed at 0x040110000000000000… on mutated_id_graph (index 1/2): {"ciphertext":"Bytes","aad":"Bytes","nonce":"[u8;12]"}:: decodeU8aStruct: failed at 0x0000… on nonce (index 3/3): [u8;12]:: Expected input with 12 bytes (96 bits), found 2 bytes
```

Reply:
:flip-table: